### PR TITLE
class progess_bar

### PR DESCRIPTION
**Pull request by bgraille**
add a progress bar class using alive_progress if possible

### CODE REVIEW

File: pylbm/__init__.py [hunk 1]
```diff
@@ -7,47 +7,52 @@
 """
 A flexible Python package for lattice Boltzmann method.
 """
+
 import logging
 from colorlog import ColoredFormatter
 import mpi4py.MPI as mpi
-
 from colorama import init
-init()
```
Comment (gouarin) [inline pylbm/__init__.py:15]: you moved the init call. Could you put it again at this place?

File: pylbm/utils.py [hunk 2]
```diff
@@ -156,3 +158,40 @@ def print_progress(iteration, total,
     if iteration == total:
         sys.stdout.write('\n')
         sys.stdout.flush()
+
+
+class my_progress_bar:
```
Comment (gouarin) [inline pylbm/utils.py:163]: Could you rename `my_progress_bar` to `pylbm_progress_bar` ?

File: pylbm/utils.py [hunk 3]
```diff
@@ -156,3 +158,40 @@ def print_progress(iteration, total,
...
+    def __exit__(self, type, value, traceback):
+        pass
+
+
+try:
+    from alive_progress import alive_bar  # , config_handler
+    # config_handler.set_global(
+    #     spinner='waves', bar='smooth'
+    # )
+    # progress_bar = alive_bar
```
Comment (gouarin) [inline pylbm/utils.py:194]: uncomment

File: pylbm/utils.py [hunk 4]
```diff
@@ -156,3 +158,40 @@ def print_progress(iteration, total,
...
+    progress_bar = my_progress_bar
+except ImportError:
+    progress_bar = my_progress_bar
```
Comment (gouarin) [inline pylbm/utils.py:197]: Could you rename my_progress_bar to pylbm_progress_bar ?

File: pylbm/__init__.py (hunk 1 above)
Comment (bgraille) [reply pylbm/__init__.py:15]: Yes but it causes a warning when using pylint. I'm deactivating it!

**Pull request merged**